### PR TITLE
Handle default active value

### PR DESCRIPTION
**Pull request by blasten**

### CODE REVIEW

File: iron-component-page.html [hunk 1]
```diff
@@ -316,28 +317,38 @@
...
-    _descriptorsChanged: function(docElements, docBehaviors) {
+    _getDefaultActive: function() {
       function findMatch(el) {
         return el && el.contentHref && el.contentHref.indexOf(mainFile) > 0;
       }
 
       var url = this._srcUrl || this.base;
       var mainFile = url.replace(_baseUrl(this.base), '');
-      var matchedPage = docElements.find(findMatch) || docBehaviors.find(findMatch);
+      var matchedPage = this.docElements.find(findMatch) || this.docBehaviors.find(findMatch);
```
Comment (frankiefu) [inline iron-component-page.html]: When I tried it I got: `Uncaught TypeError: Cannot read property 'find' of undefined iron-component-page.html:345`.  Maybe we need to check `this.docElements` and `this.docBehaviors` for null.

File: iron-component-page.html [hunk 2]
```diff
@@ -245,7 +245,8 @@
     },
 
     observers: [
-      '_descriptorsChanged(docElements, docBehaviors)'
+      '_descriptorsChanged(docElements, docBehaviors)',
+      '_frameSrc(view, base)'
```
Comment (frankiefu) [inline iron-component-page.html]: Maybe we should change to `_updateFrameSrc`.

File: iron-component-page.html (hunk 1 above)
Comment (blasten) [inline iron-component-page.html]: I set the initial value to an empty array. `docElements` and `docBehaviors` should never be null, so we don't need to check them all the time.

File: iron-component-page.html (hunk 2 above)
Comment (blasten) [inline iron-component-page.html]: changed

**Comment by blasten** [comment]
@frankiefu FYI: I had to change the approach used for to restore `history.pushState`. If you navigated back to the demo page  and pressed the back button, it was remembering the last `src` we set in the iframe.  The new approach:  create a new iframe every time you visit the section, in that way the browser won't remember the last `src`. PTAL

**Comment by frankiefu** [comment]
LGTM :+1: 

ps. we really need to rewrite this element later :-)

**Pull request merged**